api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -300,101 +299,88 @@ private DataSetRequest generateDataSetRequestFromPreviewRequest(
+       */
+      queryResponse =
+          bigQueryService.executeQuery(
+              bigQueryService.filterBigQueryConfig(queryJobConfiguration),
+              APP_ENGINE_HARD_TIMEOUT_MSEC_MINUS_FIVE_SEC);
+    } catch (Exception ex) {
+      if ((ex.getCause() != null
+          && ex.getCause().getMessage() != null
+          && ex.getCause().getMessage().contains("Read timed out"))) {
+        throw new GatewayTimeoutException(
+            "Timeout while querying the CDR to pull preview information.");
+      } else {
+        throw ex;
       }
-    } while (retry < PREVIEW_RETRY_LIMIT);
-
-    if (retry == PREVIEW_RETRY_LIMIT) {
-      throw new GatewayTimeoutException(
-          "Timeout while querying the CDR to pull preview information.");
     }
+    queryResponse
+        .getSchema()
+        .getFields()
+        .forEach(
+            fields -> {
+              valuePreviewList.add(new DataSetPreviewValueList().value(fields.getName()));
+            });
+
+    queryResponse
+        .getValues()
+        .forEach(
+            fieldValueList -> {
+              IntStream.range(0, fieldValueList.size())
+                  .forEach(
+                      columnNumber -> {
+                        try {
+                          valuePreviewList
+                              .get(columnNumber)
+                              .addQueryValueItem(
+                                  fieldValueList.get(columnNumber).getValue().toString());
+                        } catch (NullPointerException ex) {
+                          log.severe(
+                              String.format(
+                                  "Null pointer exception while retriving value for query: Column %s ",
+                                  columnNumber));
+                          valuePreviewList.get(columnNumber).addQueryValueItem("");
+                        }
+                      });
+            });
+    queryResponse
+        .getSchema()
+        .getFields()
+        .forEach(
+            fields -> {
+              DataSetPreviewValueList previewValue =
+                  valuePreviewList.stream()
+                      .filter(preview -> preview.getValue().equalsIgnoreCase(fields.getName()))
+                      .findFirst()
+                      .get();
+              if (fields.getType() == LegacySQLTypeName.TIMESTAMP) {
+                List<String> queryValues = new ArrayList<String>();
+                DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+                previewValue
+                    .getQueryValue()
+                    .forEach(
+                        value -> {
+                          try {
+                            Double fieldValue = Double.parseDouble(value);
+                            queryValues.add(dateFormat.format(new Date(fieldValue.longValue())));
+                          } catch (NumberFormatException ex) {
+                            queryValues.add("");
```

Comment:
I would log an error here, and include the value and any other context we have handy. It means the front end fed us bad information or we have a programming error on the back end. I might even make it fatal.

Reply:
Agreed. I will log an error. I don't think we want to make it fatal, because in this case some data is still valuable to a user, but we should log something in our system

Reply:
Discussed offline, and this is just a try/catch used in place of a null check. Just a style/readability problem.